src/term/transform/scott_generation.rs

```diff
@@ -0,0 +1,48 @@
+use crate::term::{Book, Definition, Name, Rule, Term};
+
+impl Book {
+  pub fn generate_scott_adts(&mut self) {
+    for adt in self.adts.values() {
+      let curr_ctrs = adt.ctrs.clone();
+
+      for (ctr_name, arity) in &curr_ctrs {
+        let ctrs = curr_ctrs.keys().cloned().collect();
+        let ctr_args = make_args(arity);
+
+        let lam = make_lam(ctr_args, ctrs, ctr_name);
+
+        let def_id = self.def_names.insert(ctr_name.clone());
+
+        let rules = vec![Rule { def_id, pats: vec![], body: lam }];
+        let def = Definition { def_id, rules };
+
+        self.defs.insert(def_id, def);
+        self.ctrs.remove(ctr_name);
```

Comment:
Are you sure we wont' use this for other things kater? Seems unwise to remove it here

Reply:
I was probably testing something and I forgot to remove this line, my fault